src/Tizen.Applications.ComponentBased/Tizen.Applications.ComponentBased.Common/ComponentBasedApplication.cs

```diff
@@ -0,0 +1,209 @@
+            }
+            else
+            {
+                throw new ArgumentException("Invalid type");
+            }
+        }
+
+        /// <summary>
+        /// Runs the application's main loop.
+        /// </summary>
+        /// <param name="args">Arguments from commandline.</param>
+        /// <exception cref="InvalidOperationException">Thrown when component type is already added to the component.</exception>
+        /// <since_tizen> 6 </since_tizen>
+        public override void Run(string[] args)
+        {
+            base.Run(args);
+
+            string[] argsClone = new string[args.Length + 1];
+            if (args.Length > 1)
+            {
+                args.CopyTo(argsClone, 1);
+            }
+            argsClone[0] = string.Empty;
+
+            Interop.CBApplication.ErrorCode err = Interop.CBApplication.BaseMain(argsClone.Length, argsClone, ref _callbacks, IntPtr.Zero);
+            if (err != Interop.CBApplication.ErrorCode.None)
+            {
+                Log.Error(LogTag, "Failed to run the application. Err = " + err);
+                throw new InvalidOperationException("Fail to run application : err(" + err + ")");
+            }
+        }
+
+        /// <summary>
+        /// Exits the main loop of the application.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public override void Exit()
+        {
+            Interop.CBApplication.BaseExit();
+        }
+
+        private IntPtr OnCreateNative(IntPtr data)
+        {
+            Log.Debug(LogTag, "On create");
+            IntPtr h = IntPtr.Zero;
+            foreach(KeyValuePair<Type, ComponentFactoryBase> entry in _componentFactories)
+            {
+                h = entry.Value.Bind(h);
+            }
+
+            return h;
+        }
+
+        private void OnTerminateNative(IntPtr data)
+        {
+        }
+
+        private void OnRunNative(IntPtr data)
+        {
+            OnRun();
+        }
+
+        private void OnExitNative(IntPtr data)
+        {
+            OnExit();
+        }
+
+        private void OnInitNative(int argc, string[] argv, IntPtr userData)
+        {
+            OnInit(argv);
+        }
+
+        private void OnFinishedNative(IntPtr data)
+        {
+            OnFinished();
+        }
+
+        /// <summary>
+        /// This method will be called before running main-loop
+        /// </summary>
+        /// <param name="args"></param>
+        /// <since_tizen> 6 </since_tizen>
+        protected virtual void OnInit(string[] args)
+        {
+        }
+
+        /// <summary>
+        /// This method will be called after exiting main-loop
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        protected virtual void OnFinished()
+        {
+        }
+
+        /// <summary>
+        /// This method will be called to start main-loop
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        protected virtual void OnRun()
```

Comment:
It should be abstract, 
because if not override this method, application is not working

Reply:
Done